libcontainer/rootfs_linux.go

```diff
@@ -76,6 +77,13 @@ func prepareRootfs(pipe io.ReadWriter, config *configs.Config) (err error) {
 		}
 	}
 
+	// Setup Cwd before running the pre-start hooks, so that hooks can play with Cwd.
```

Comment:
why not do this right before chdir to the cwd when the rootfs is already setup and every other mount is done and we are jailed inside the rootfs?

Reply:
That's what I was about to comment :wink:. That way we don't need to use `CleanPath` we can just use `MkdirAll` and nothing else. If a hook wants to mess with the cwd, they could create it themselves.